embassy-mspm0/src/wwdt.rs

```diff
@@ -324,6 +400,16 @@ impl Watchdog {
     }
 }
 
+/// Read the reset cause from the SYSCTL.RSTCAUSE register.
+/// 
+/// This function reads the reset cause register which indicates why the last
+/// system reset occurred. The register is automatically cleared after being read,
+/// so this should be called only once per application startup.
+pub fn read_reset_cause() -> Result<ResetCause, u8> {
+    let cause_raw = pac::SYSCTL.rstcause().read().id();
+    ResetCause::try_from(cause_raw as u8)
+}
+
```

Comment:
This does open up an interesting safety question: Should be allow the user to call this more than once? With peripheral init it makes sense to do that since otherwise memory safety errors would happen.

In this case it would just produce the wrong reset value. At minimum probably a good idea to mark this function as `#[must_use = "Reading reset cause will clear it"]` so that the compiler can warn if you try to use it and ignore the result.